assets/javascripts/discourse/services/chat-emoji-store.js

```diff
@@ -0,0 +1,57 @@
+import KeyValueStore from "discourse/lib/key-value-store";
+import Service from "@ember/service";
+
+const EMOJI_USAGE = "emojiUsage";
+const EMOJI_SELECTED_DIVERSITY = "emojiSelectedDiversity";
+const TRACKED_EMOJIS = 15;
+const STORE_NAMESPACE = "discourse_chat_emojis_";
+
+export default Service.extend({
```

Comment:
I wonder if we could just import the `EmojiStore` service from core here and do `Service.extend(EmojiStore)`? That would save having to copy all this stuff from core

Reply:
Oh I see that `get reactions()` is new here too, maybe like this?

```javascript
Service.extend(EmojiStore, {
   get reactions() {
    if (!this.siteSettings.default_emoji_reactions) {
      return [];
    }
    return this.siteSettings.default_emoji_reactions.split("|").filter(Boolean);
  },
})
```

Reply:
Thanks for the suggestion! I tried doing this but it seems like it's not possible to import `EmojiStore` service from core.

Reply:
Yep no problem just need the comment in this class explaining it's duplicated from core for future devs

Reply:
Done.